File: server/datasets/python_source-code/Multiple-E3/organic_prefix_sum_1_max_subarray.py
def maxSubArray(nums):
    max_sum = float('-inf')
    current_prefix = 0
    min_prefix = 0
    
    for num in nums:
        current_prefix += num
        if current_prefix - min_prefix > max_sum:
            max_sum = current_prefix - min_prefix
        if current_prefix < min_prefix:
            min_prefix = current_prefix
            
    return max_sum

File: server/datasets/controlled_benchmarks/java/multiple_files/multiple_e3/organic_prefix_sum_1_max_subarray.java
public class Solution {
    public int maxSubArray(int[] nums) {
        int max_sum = Integer.MIN_VALUE;
        int current_prefix = 0;
        int min_prefix = 0;
        
        for (int num : nums) {
            current_prefix += num;
            if (current_prefix - min_prefix > max_sum) {
                max_sum = current_prefix - min_prefix;
            }
            if (current_prefix < min_prefix) {
                min_prefix = current_prefix;
            }
        }
        return max_sum;
    }
}
